src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py

```diff
@@ -74,3 +75,18 @@ def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
             compression_bundle = dict(compression=self.compression_method, compression_opts=self.compression_options)
 
         return dict(chunks=self.chunk_shape, **compression_bundle)
+
+
+class HDF5BackendConfiguration(BackendConfiguration):
+    """A model for matching collections of DatasetConfigurations specific to the HDF5 backend."""
+
+    backend: Literal["hdf5"] = Field(
+        default="hdf5", description="The name of the backend used to configure the NWBFile."
+    )
```

Comment:
Wouldn't HDF5BackendConfiguration already imply that the backend is hdf5? It looks like you are making it possible to set to some other value

Reply:
I'm usings `Field` because that enhances the JSON schema output of the Pydantic model, which is what the GUIDE would use to make those little info icons

The `default` of a `Field` is equivalent syntax to `variance_name = "default value"`

Otherwise largely for consistency with the parent class where the choice is between string options "hdf5" and "zarr"

Though I will note that the workflow used later on would rarely have users instantiate these objects themselves - the convenience functions return it for them, so most of this is just for internal dataclass consistency

Reply:
Datclasses have `init=False` to explichttps://docs.python.org/3/library/dataclasses.html#dataclasses.fieldtly indicate that something is not mean to be passed to the `__init__` method:

https://docs.python.org/3/library/dataclasses.html#dataclasses.field

Doesn't pydantic support something similar?

Reply:
Easily, by simply overriding the `__init__` to exactly specify the signature. Would you prefer that?

The point is that these are more class attributes than they are initialized values of the dataclass, though once you have an instance it wouldn't matter either way